collie/model/hybrid_matrix_factorization.py

```diff
@@ -213,36 +237,98 @@ def __init__(self,
                     'lr': metadata_only_stage_lr,
                     'optimizer': metadata_only_stage_optimizer,
                     # optimize metadata layers only
-                    'parameter_prefix_list': ['metadata', 'combined', 'user_bias', 'item_bias'],
+                    'parameter_prefix_list': [
+                        'item_metadata', 'user_metadata', 'combined', 'user_bias', 'item_bias'
+                    ],
                     'stage': 'metadata_only',
                 },
                 {
                     'lr': all_stage_lr,
                     'optimizer': all_stage_optimizer,
                     # optimize everything
-                    'parameter_prefix_list': ['user', 'item', 'metadata', 'combined'],
+                    'parameter_prefix_list': [
+                        'user', 'item', 'item_metadata', 'user_metadata', 'combined'
+                    ],
                     'stage': 'all',
                 },
             ]
 
         super().__init__(optimizer_config_list=optimizer_config_list,
                          item_metadata_num_cols=item_metadata_num_cols,
+                         user_metadata_num_cols=user_metadata_num_cols,
                          **get_init_arguments())
 
     __doc__ = merge_docstrings(MultiStagePipeline, __doc__, __init__)
 
     def _move_any_external_data_to_device(self):
-        """Move item metadata to the device before training."""
-        self.item_metadata = self.item_metadata.to(self.device)
+        """Move item and user metadata to the device before training."""
+        if self.item_metadata is not None:
+            self.item_metadata = self.item_metadata.to(self.device)
+        if self.user_metadata is not None:
+            self.user_metadata = self.user_metadata.to(self.device)
 
     def _load_model_init_helper(self, load_model_path: str, map_location: str, **kwargs) -> None:
-        self.item_metadata = (
-            joblib.load(os.path.join(load_model_path, 'metadata.pkl'))
-        )
+        try:
+            self.item_metadata = (
+                joblib.load(os.path.join(load_model_path, 'item_metadata.pkl'))
+            )
+        except FileNotFoundError:
+            if self.hparams.item_metadata_layers_dims is not None:
+                warnings.warn('```item_metadata.pkl`` not found')
+        try:
+            self.user_metadata = (
+                joblib.load(os.path.join(load_model_path, 'user_metadata.pkl'))
+            )
+        except FileNotFoundError:
+            if self.hparams.user_metadata_layers_dims is not None:
+                warnings.warn('``user_metadata.pkl`` not found')
+
         super()._load_model_init_helper(load_model_path=os.path.join(load_model_path, 'model.pth'),
                                         map_location=map_location,
                                         **kwargs)
 
+    def _configure_metadata_layers(
```

Comment:
@nathancooperjones This is your version of the `_configure_metadata_layers` function, I wrote a slightly different version in the `pretrained` model and wanted you to compare and see if you have a preference